src/commcare_cloud/ansible/roles/ecryptfs/tasks/main.yml

```diff
@@ -1,119 +1,121 @@
-  tags:
-    - mount-ecryptfs
-    - after-reboot
+    - name: Add encryption password to keyring
+      become: yes
+      shell: 'printf "%s" "{{ ECRYPTFS_PASSWORD }}" | ecryptfs-add-passphrase | grep -o "\[.*\]" | sed "s/\[//g;s/\]//g"'
+      register: password_hash
+      when: encrypted_root  not in mtab_contents.stdout
+      tags:
+        - mount-ecryptfs
+        - after-reboot
 
-- name: Mount data drive
-  become: yes
-  shell: "mount -t ecryptfs -o key=passphrase:passphrase_passwd={{ ECRYPTFS_PASSWORD }},user,noauto,ecryptfs_cipher=aes,ecryptfs_key_bytes=32,ecryptfs_unlink_sigs,ecryptfs_enable_filename_crypto=y,ecryptfs_fnek_sig={{ password_hash.stdout }},verbosity=0 {{ encrypted_root }}/ {{ encrypted_root }}/"
-  when: (password_hash.stdout is defined) and (encrypted_root not in mtab_contents.stdout)
-  tags:
-    - mount-ecryptfs
-    - after-reboot
+    - name: Mount data drive
+      become: yes
+      shell: "mount -t ecryptfs -o key=passphrase:passphrase_passwd={{ ECRYPTFS_PASSWORD }},user,noauto,ecryptfs_cipher=aes,ecryptfs_key_bytes=32,ecryptfs_unlink_sigs,ecryptfs_enable_filename_crypto=y,ecryptfs_fnek_sig={{ password_hash.stdout }},verbosity=0 {{ encrypted_root }}/ {{ encrypted_root }}/"
+      when: (password_hash.stdout is defined) and (encrypted_root not in mtab_contents.stdout)
+      tags:
+        - mount-ecryptfs
+        - after-reboot
 
-- name: "Create {{ encrypted_root }}/blobdb directory for setups that do not use NFS"
-  become: yes
-  when: not shared_drive_enabled
-  file: path="{{ encrypted_root }}/blobdb" owner=nobody group="{{ shared_dir_gid }}" mode=0775 state=directory
-  tags:
-    - ecryptfs-blobdb
+    - name: "Create {{ encrypted_root }}/blobdb directory for setups that do not use NFS"
+      become: yes
+      when: not shared_drive_enabled
+      file: path="{{ encrypted_root }}/blobdb" owner=nobody group="{{ shared_dir_gid }}" mode=0775 state=directory
+      tags:
+        - ecryptfs-blobdb
 
-- name: Create shared data dir
-  become: yes
-  file:
-    path: "{{ shared_data_dir }}"
-    owner: "nobody"
-    group: "{{ shared_dir_gid }}"
-    mode: "u=rwx,g=rwx,o=rx"
-    state: directory
-  when: not shared_drive_enabled
-  tags:
-    - ecryptfs-blobdb
+    - name: Create shared data dir
+      become: yes
+      file:
+        path: "{{ shared_data_dir }}"
+        owner: "nobody"
+        group: "{{ shared_dir_gid }}"
+        mode: "u=rwx,g=rwx,o=rx"
+        state: directory
+      when: not shared_drive_enabled
+      tags:
+        - ecryptfs-blobdb
 
-- name: Create blob DB symlink (cchq looks for it in the shared drive)
-  become: yes
-  when: not shared_drive_enabled
-  file: src="{{ encrypted_root }}/blobdb" dest="{{ shared_data_dir }}/blobdb" state=link
-  tags:
-    - ecryptfs-blobdb
+    - name: Create blob DB symlink (cchq looks for it in the shared drive)
+      become: yes
+      when: not shared_drive_enabled and root_encryption_mode != 'aws'
```

Comment:
It looks like this was left accidentally?

Reply:
yes Danny and removed